### PR TITLE
feat: settings sliders

### PR DESCRIPTION
**Pull request by MeguminSama**
Adds a Slider component type to plugin settings.

Also removes some unsafe destructuring of proxies in other component files.

### CODE REVIEW

File: src/components/PluginSettings/components/SettingSliderComponent.tsx [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import { ISettingElementProps } from ".";
+import { PluginOptionSlider } from "../../../utils/types";
+import { Forms, React, Slider } from "../../../webpack/common";
+
+export function makeRange(start: number, end: number, step = 1) {
+    let ranges: number[] = [];
```
Comment (Vendicated) [inline src/components/PluginSettings/components/SettingSliderComponent.tsx:6]:
```suggestion
    const ranges: number[] = [];
```

File: src/components/PluginSettings/components/SettingSliderComponent.tsx [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+    for (let value = start; value <= end; value += step) {
+        ranges.push(parseFloat(value.toFixed(2)));
```
Comment (Vendicated) [inline src/components/PluginSettings/components/SettingSliderComponent.tsx:8]:
```suggestion
        ranges.push(Math.round(value * 100) / 100));
```

File: src/webpack/common.tsx [hunk 3]
```diff
@@ -31,6 +31,7 @@ export let TextInput: any;
 export let Text: (props: TextProps) => JSX.Element;
 
 export const Select = lazyWebpack(filters.byCode("optionClassName", "popoutPosition", "autoFocus", "maxVisibleItems"));
+export const Slider = lazyWebpack(filters.byCode("closestMarkerIndex", "stickToMarkers"));
```
Comment (Vendicated) [inline src/webpack/common.tsx:34]: btw discord-types may have types for these

**Review by Vendicated — CHANGES_REQUESTED** [review]

File: src/webpack/common.tsx (hunk 3 above)
Comment (MeguminSama) [reply src/webpack/common.tsx:34]: the Slider types in discord-types are very incomplete.

**Review by Vendicated — APPROVED** [review]

**Pull request merged**